jlm/llvm/opt/alias-analyses/PointerObjectSet.cpp

```diff
@@ -1458,6 +1484,17 @@ PointerObjectConstraintSet::RunWorklistSolver(WorklistStatistics & statistics)
     callConstraints[root].UnionWith(callConstraints[nonRoot]);
     callConstraints[nonRoot].Clear();
 
+    if constexpr (EnableHybridCycleDetection)
+    {
+      // When unifying, keep at least one of the ref node unification roots
```

Comment:
The comment indicates that at least one root should ALWAYS be kept. The code below does not enforce this though. Could it still be that `RefNodeUnificationRoot_[root].count(root)` is 0 after this piece of code? If not, maybe it would be possible to put an assert there?

Reply:
I changed the comment to make it clearer. It is possible that neither the root or the root's new child have their ref node in a cycle with a regular node